teps/0061-allow-custom-task-to-be-embedded-in-pipeline.md

```diff
@@ -112,9 +112,177 @@ be handled, or user scenarios that will be affected and must be accomodated.
+  name: simpletasklooprun
+spec:
+  params:
+    - name: word
+      value:
+        - jump
+        - land
+        - roll
+    - name: suffix
+      value: ing
+  spec:
+    apiVersion: custom.tekton.dev/v1alpha1
+    kind: TaskLoop
+    spec:
+      # Task to run (inline taskSpec also works)
+      taskRef:
+        name: simpletask
+      # Parameter that contains the values to iterate
+      iterateParam: word
+      # Timeout (defaults to global default timeout, usually 1h00m; use "0" for no timeout)
+      timeout: 60s
+      # Retries for task failure
+      retries: 2
+```
+
+Another example based on `PipelineRun` spec, will look like:
+
+_Note that, `spec.pipelineSpec.tasks.taskSpec.spec` is holding the custom task spec._
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+ kind: PipelineRun
+ metadata:
+   name: pr-loop-example
+ spec:
+   pipelineSpec:
+     tasks:
+       - name: first-task
+         taskSpec:
+           steps:
+             - name: echo
+               image: ubuntu
+               imagePullPolicy: IfNotPresent
+               script: |
+                 #!/usr/bin/env bash
+                 echo "I am the first task before the loop task"
+       - name: loop-task
+         runAfter:
+           - first-task
+         params:
+           - name: message
+             value:
+               - I am the first one
+               - I am the second one
+               - I am the third one
+         taskSpec:
+           apiVersion: custom.tekton.dev/v1alpha1
+           kind: PipelineLoop
+           spec:
+             iterateParam: message
+             pipelineSpec:
+               params:
+               - name: message
+                 type: string
+               tasks:
+               - name: echo-loop-task
+                 params:
+                 - name: message
+                   value: $(params.message)
+                 taskSpec:
+                   params:
+                   - name: message
+                     type: string
+                   steps:
+                     - name: echo
+                       image: ubuntu
+                       imagePullPolicy: IfNotPresent
+                       script: |
+                         #!/usr/bin/env bash
+                         echo "$(params.message)"
+       - name: last-task
+         runAfter:
+           - loop-task
+         taskSpec:
+           steps:
+             - name: echo
+               image: ubuntu
+               imagePullPolicy: IfNotPresent
+               script: |
+                 #!/usr/bin/env bash
+                 echo "I am the last task after the loop task"
+```
+
+`Tektoncd/pipeline` can only validate the structure and fields it knows about, validation of the
+custom task spec field(s) is delegated to the custom task controller.
+
+A custom controller may still choose to not support one of `Spec` based `Run` specification or
+`Ref` based ones. This can be done by implementing suitable webhooks at the custom controller end.
```

Comment:
Just to clarify this a bit more: what kind of effect does this have on the PipelineRun?  Would it have to wait for its timeout deadline or would it be able to immediately detect the error and be placed into a Succeeded/False condition?

Reply:
Currently with taskRef, if the custom task controller did not reconcile the `Run` CRD, then PipelineRun just keep waiting  until its timeout deadline. We will follow the same behavior for taskSpec (Run Spec) as well.

@ScrapCodes can you update the doc to clarify this behavior? thanks

Reply:
I just updated the doc.